### PR TITLE
Feature confirm appointment #13

### PR DESCRIPTION
**Pull request by LKhyeon**
### What this PR does (required):
Modifications to the url of the scheduler.
- When a time slot is clicked, the app prompt a confirmation message. Clicking on the continue button takes the user to the demo completion page, but when integrating, it will create an appointment and send the user to `completion/[appointment id]`.
- When the user continues, the app takes the user to the completion page. All of the information shown in the page are dummy data. When integrating, appointment model with the id from the url will be used to populate all the data.
- Clicking on reschedule will take the user back to the scheduler. When integrating, it will also remove the corresponding appointment model.

### Screenshots / Videos (required):
- New scheduler url of the form `/scheduler/[username]/[duration]`.
image

- Confirmation message when the time slot is clicked.
image

- Sample of the completion page. 
image

### Any information needed to test this feature (required):
- Since there is no legitimate way to get to the scheduler (since dashboard integration is not complete), edit the app.tsx such that `<AuthProvider>` is not being used.
image

- `cd /client` -> run `npm install` -> run `npm start`
- `cd /server `-> run `npm install` -> run `npm run dev`
- Go to localhost `http://localhost:3000/scheduler/TEMP/30` (feel free to replace TEMP with some other string and 60 with some other number).

### Any issues with the current functionality (optional):
- N/A

### CODE REVIEW

File: client/src/interface/SchedulerProps.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+export interface completionUrlProp {
+  appointID: string;
+}
+
+export interface confirmProp {
+  username: string;
+  duration: number;
+  timeZone: string;
+  time: Moment;
+  // checkConfirmation: clickEventHandler;
```
Comment (moffatethan) [inline client/src/interface/SchedulerProps.ts]: Is this not needed anymore?

Comment (LKhyeon) [reply client/src/interface/SchedulerProps.ts]: Yes, it is not needed anymore. I should have removed it but I missed it.

File: client/src/pages/Scheduler/Scheduler.tsx [hunk 2]
```diff
@@ -99,12 +105,23 @@ export default function Scheduler(): JSX.Element {
...
+        <Confirmation
+          username={username}
+          duration={Number(duration)}
+          timeZone={timeZone}
+          time={moment.tz(dateSelISO, timeZone)}
+          cancelConfirmation={cancelConfirmation}
+        />
+      ) : (
+        <Box />
+      )}
```
Comment (moffatethan) [inline client/src/pages/Scheduler/Scheduler.tsx]:
```suggestion
      {confirmTrigger && (
        <Confirmation
          username={username}
          duration={Number(duration)}
          timeZone={timeZone}
          time={moment.tz(dateSelISO, timeZone)}
          cancelConfirmation={cancelConfirmation}
        />
      )}
```

**Review by moffatethan — COMMENTED** [review]
Left you some comments but good work!

File: client/src/pages/Scheduler/Scheduler.tsx (hunk 2 above)
Comment (LKhyeon) [reply client/src/pages/Scheduler/Scheduler.tsx]: I forgot I am able to do this in react. Thank you!

**Review by moffatethan — APPROVED** [review]
LGTM!

**Pull request merged**